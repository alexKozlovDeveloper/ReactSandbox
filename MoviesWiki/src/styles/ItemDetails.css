.body {
    background-color: tomato;
    display: flex;
    padding-left: 30px;
    padding-right: 30px;
    padding-top: 10px;
    padding-bottom: 10px;
}

.imagecontainer {

}

.detailscontainer {
    padding: 10px;
}

.maintitlecontainer {
    display: flex;
}

.propertiescontainer {
    display: flex;
}

.descriptioncontainer {
    
}