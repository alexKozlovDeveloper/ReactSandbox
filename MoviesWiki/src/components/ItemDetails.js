import React, { Component } from "react";
import { connect } from 'react-redux'

import ItemImage from "./Common/ItemImage";
import ItemDetailsTitle from "./Common/ItemDetailsTitle";
import ItemRating from "./Common/ItemRating";
import Description from "./Common/Description";
import ItemYearAndDuration from "./Common/ItemYearAndDuration";
import BackButton from "./Common/BackButton";

import styles from "../styles/ItemDetails.css"


class ItemDetails extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        debugger;
        return (
            <div className={styles.body}>
                <div className={styles.imagecontainer}>
                    <ItemImage image={this.props.item.poster_path} />
                </div>
                <div className={styles.detailscontainer}>
                    <div className={styles.maintitlecontainer}>
                        <ItemDetailsTitle title={this.props.item.title} />  
                        <ItemRating rating={this.props.item.vote_average} />     
                    </div>
                    <div className={styles.propertiescontainer}>
                        <ItemYearAndDuration year={this.props.item.release_date} duration={this.props.item.runtime}/>
                    </div>
                    <div className={styles.descriptioncontainer}>
                        <Description description={this.props.item.overview} />   
                    </div>  
                </div>  
                <div>
                    <BackButton buttonText=""/>
                </div> 
            </div>
        );   
    } 
}

function mapStateToProps(state){    
    const { selectedItem } = state.moviesReducer;

    return { 
        item: selectedItem
    };
}

export default connect(mapStateToProps)(ItemDetails);
