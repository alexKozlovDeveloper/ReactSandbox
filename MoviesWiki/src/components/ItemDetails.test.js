import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import rootReducer from '../reducers'
import { createStore } from 'redux'
import { Provider } from 'react-redux'

import ItemDetails from "./ItemDetails";

configure({adapter: new Adapter()});

test('ItemDetails rendering test', () => {

    var item = {
        title: "Interstellar 4",
        releaseDate: "2017",
        image: "https://m.media-amazon.com/images/M/MV5BZjdkOTU3MDktN2IxOS00OGEyLWFmMjktY2FiMmZkNWIyODZiXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SY1000_SX675_AL_.jpg",        
        genre: "action 4",
        rating: 7.2,
        duration: 145,
        description: "is a 2014 British-American epic science fiction film directed and produced by Christopher Nolan. ... Set in a dystopian future where humanity is struggling to survive, the film follows a group of astronauts who travel through a wormhole near Saturn in search of a new home for humanity.",
    }

    var defaultStore = {
        moviesReducer: {
            selectedItem: item
        }
    }

    const store = createStore(rootReducer, defaultStore)

    const component = shallow(<Provider store={store}>
                                <ItemDetails item={item}/>
                              </Provider>);
    expect(shallowToJson(component)).toMatchSnapshot();
 });

