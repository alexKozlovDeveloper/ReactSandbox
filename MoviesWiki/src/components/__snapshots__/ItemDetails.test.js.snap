// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemDetails rendering test 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(ItemDetails)
    item={
      Object {
        "description": "is a 2014 British-American epic science fiction film directed and produced by Christopher Nolan. ... Set in a dystopian future where humanity is struggling to survive, the film follows a group of astronauts who travel through a wormhole near Saturn in search of a new home for humanity.",
        "duration": 145,
        "genre": "action 4",
        "image": "https://m.media-amazon.com/images/M/MV5BZjdkOTU3MDktN2IxOS00OGEyLWFmMjktY2FiMmZkNWIyODZiXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SY1000_SX675_AL_.jpg",
        "rating": 7.2,
        "releaseDate": "2017",
        "title": "Interstellar 4",
      }
    }
  />
</ContextProvider>
`;
